src/Entity/DatasetSubmission.php

```diff
@@ -1155,6 +1166,18 @@ public function __construct(Entity $entity, PersonDatasetSubmissionDatasetContac
                 $newDistributionPoint->setDataCenter($distributionPoint->getDataCenter());
                 $this->addDistributionPoint($newDistributionPoint);
             }
+
+            // Copy the original Dataset Submission's Dataset Links.
+            foreach ($entity->getDatasetLinks() as $datasetLink) {
+                $newDatasetLink = new DatasetLink();
+                $newDatasetLink->setUrl($datasetLink->getUrl());
```

Comment:
If we are going to use return types for setters, then we should use method chaining as well! I think if aren't going to use method chaining, we shouldn't use return type for setters.

For example:

`class fakeString
{
    private $str;
    function __construct()
    {
        $this->str = "";
    }

    function addA()
    {
        $this->str .= "a";
        return $this;
    }

    function addB()
    {
        $this->str .= "b";
        return $this;
    }

    function getStr()
    {
        return $this->str;
    }
}


$a = new fakeString();


echo $a->addA()->addB()->getStr();`

Reply:
How would you improve this particular scenario?

Reply:
````
$newDatasetLink->setUrl($datasetLink->getUrl())->setName($datasetLink->getName())->setDescription($datasetLink->getDescription()) ......
````
We can do this, but I feel chaining methods like this is tough to understand

Reply:
I'm actually ok with them too, but think we should decide one way or another and stick with it.